packages/web-components/src/components/va-modal/va-modal.tsx

```diff
@@ -132,6 +132,11 @@ export class VaModal {
    */
   @Prop({ reflect: true }) visible?: boolean = false;
 
+  /**
+   * DOM-nodes that should not be hidden from screen readers.
+   */
+  @Prop() ariaHiddenNodeExceptions?: HTMLElement[] = [];
```

Comment:
@bkjohnson I wonder if you had any thoughts about this issue. The component docs created via `stencil build --docs-json component-docs.json` don't generate this as a property for displaying in storybook:

![Screen Shot 2022-10-26 at 11 26 14 AM](https://user-images.githubusercontent.com/872479/198082180-fcf8d432-6ea0-4905-a555-717499358259.png)

Because it's an array type, it won't ever generate with an attribute name (because it's a property not an attribute) and therefore won't display correctly in our Storybook for it.

Using this new attribute to solve the screenreader issue seemed like the cleanest approach but I'm sure how to solve for it showing up in Storybook like this.

Reply:
Maybe we switch `prop.attr` here to be `prop.name`, or do something like `prop.attr || prop.name`:

https://github.com/department-of-veterans-affairs/component-library/blob/ce3b4b70118991bf2b380aac7f498457427a4cb4/packages/storybook/stories/wc-helpers.jsx#L85

![image](https://user-images.githubusercontent.com/2008881/198148893-8d4c17fa-5526-48f7-8d28-9114081e2ec3.png)


I couldn't find any instances where we are [setting the `attribute` inside the `@Prop` decorator](https://stenciljs.com/docs/properties#attribute-name-attribute), so in general it looks like the main difference between `name` and `attr` in the JSON docs is camel casing vs kebab casing:

![image](https://user-images.githubusercontent.com/2008881/198149355-0a6036e9-0707-400b-95f1-ed613e196266.png)

Reply:
Yes, good suggestion. I can make that update. I wonder though if there are any implications of having an array prop like that though (seeing as how this is the first time we've needed it).

Reply:
The array prop will mean that it can't (easily) be used in plain HTML the same way that booleans, strings, and numbers can be used. The `<va-search-input>` component is similar with its `suggestions` array prop (that's why the story uses a react binding).

https://github.com/department-of-veterans-affairs/component-library/blob/ce3b4b70118991bf2b380aac7f498457427a4cb4/packages/storybook/stories/va-search-input.stories.jsx#L98

It's like if instead of just being able to do [this](https://github.com/department-of-veterans-affairs/content-build/blob/6dc80d182b656f43e837d0292ad79be698750a9c/src/site/components/informational_alert.drupal.liquid#L2):

```html
  <va-alert status="info" id="va-info-alert">
    <h3 slot="headline">
      Need help after hours?
    </h3>
    <div>
      {{ entity.fieldWysiwyg.0.processed | drupalToVaPath | phoneLinks}}
    </div>
  </va-alert>
```

We had to have an additional `<script>` that did this if we wanted to set the `status` prop:

```js
const alertEl = document.getElementById('va-info-alert');
alertEl.status = "info";
```

Except for the modal situation we'd have to have an array of DOM nodes instead of strings, which would mean even more JS in that extra script.